src/Sentry/TransactionTracer.cs

```diff
@@ -313,9 +313,16 @@ private void AddChildSpan(SpanTracer span)
         if (!isOutOfLimit)
         {
             _spans.Add(span);
+            _sortedSpanCache = null; // invalidate cache
         }
     }
 
+    private IOrderedEnumerable<ISpan>? _sortedSpanCache;
+    /// <inheritdoc />
+    public ISpan? GetLastActiveSpan() =>
+        // We need to sort by timestamp because the order of ConcurrentBag<T> is not deterministic
+        (_sortedSpanCache ??= Spans.OrderByDescending(x => x.StartTimestamp)).FirstOrDefault(s => !s.IsFinished);
```

Comment:
Can we replace these LINQ usages with loops? That'll avoid allocations too

Reply:
Also: Isnt' TransactionTrace supposed to be thread safe? In a GlobalMode=true scenario, multiple threads can be adding spans. So iterating on this collection will throw at runtime.

Reply:
Replacing it with a for loop gave even better results... sometimes I guess it's good to go back to basics!